### ISSUE TITLE
Grey screen in some games with multiple-input driver

### ISSUE THREAD

**Issue by hestenet** [pushrax/OpenVR-SpaceCalibrator#18]
Firstly, thank you for building the Space-Calibrator. It's been awesome being able to use the Index Controllers with my Odyssey+. 

I do experience issues in some games, and I'm not 100% sure if it's the OpenVR-Space Calibrator or just an error caused by multiple-input drivers in SteamVR. 

Basically - the game starts up as normal, audio works properly, the mirror screen is enabled, but the screen on the HMD is just greyed out. The mirror screen does not show any controllers floating in space, and does not turn the viewport with the headset. 

One example game is Polynomial 2. Although I also seem to see this with some Revive games (i.e: Borderlands 2 VR, with Revive to take advantage of oculus style controller maps). 

Any ideas as to what's causing this? Would love to be able to use my hybrid-VR setup in some of these other games.

**Comment by hestenet** [comment]
My suspicion is some kind of mismatch where the game is assuming the headset based on the type of controller it sees, instead of the other way around. I'm not sure if it's possible to hijack the way steamvr reports what inputs are connected to trick it into believe it has the right one?

**Comment by hestenet** [comment]
In case it is a Revive issue or SteamVR issue, I've also opened issues in those queues: 
https://github.com/LibreVR/Revive/issues/1531
https://steamcommunity.com/app/250820/discussions/0/2266942917228846120/

**Comment by Alde1980** [comment]
Hi I habe this Issue with DCS world. I have tried steam client and non-steam client.
My setuo is using a Hp Reverb WMR and vive controllers. Would be great to solve this issue. I know for sure the gray screen appears when HMD isnt tracking, I have read about it in an steamVR thread and modified a config file so tje screen is no longer gray, you can sed the image but the hmd nor the controllers are tracked.

**Comment by Alde1980** [comment]
Hi I solved my DCS not tracking/Gray screen with multiple drivers!


I take the idea from hestenet assumption about a mismatch with contollers and tracker.
What I have done:
-After had my spaces calibrated with OpenSpace calibrator I made a lighthouse setup with two vive controllers and a third one converted to a vive tracker using a steam controller dongle (followed this tutorial: https://youtu.be/8ykyS447OF4) 
I assume that using a real tracker would work
-Turned OFF my WMR controllers
-I then exit from steam vr and WMR portal.
-Attached fisically tracker to Hp Reverb WMR headset
-Started WMR portal (DIDNT TURN ON WMR CONTROLLERS)
-Started WMR for steam VR from Steam
-Once inside SteamVR using OpenVR input emulator software set the tracker to REDIRECT to HMD.
this way I have two vive controllers and the tracking is done with Vive tracker, so full lighthowse.
Then I started DCS and it worked flawesly.

Hope this helps someone and helps track the origin of the gray screen issue

**Comment by 4dos** [comment]
I have nearly the same problem, every game (without H.L.:A) does not start. First the start screen is shown and when I start the "knuckles" the screen turns gray. Low-Fi, covert phantom operations, Vox Machinae, the dark room, a.s.o. I think this is a common problem, and if the new HP Reverb G2 sells really well, the request will be a lot more.

**Comment by pushrax** [comment]
There’s something different about the way WMR handles space setup that seems to break sometimes. WMR is the one platform I don’t have an HMD to test with, so it’s been difficult to track down.

Any contributions related to fixing this are welcome.

**Comment by ithirzty** [comment]
So is there no solution?

**Comment by JstuffJr** [comment]
@pushrax I can't believe I didn't notice you commented here earlier. 

I extensively use your application daily for development and play with HP Reverb + Knuckles, and I've posted all over reddit educating people how to  use dongles + OVRSC to get the best tracking working in their non lighthouse HMD. With the G2 release imminent, knuckles + G2 should be a (comparatively) popular combo I'm sure a lot more people will be using vs the current userbase. 

I am quite confident from 10s of hours of testing that the gray screen issue is in any steamvr application that uses a steamvr seated-mode tracking space. As @Alde1980 says, you can use OpenVR Input Emulator to redirect a vive tracker to turn the entire system into native lighthouse to get around this, but unfortunately OVRIE is completely abandoned and breaks both with Windows 2004, and the SteamVR naming change you recently had to update for. 

Since the devs of OpenVR Advanced Settings have said they don't have interesting in facilitating lighthouse + WMR, this leaves your application as the sole actively maintained software solution. Because it is so valuable to me on a daily basis (in addition to all the others who will soon be using it), I'm willing to literally ship you my spare HP Reverb for free once my Reverb G2 arrives next month, if you can reasonably commit to using it to support WMR.

Again, I'm confident it is only when using WMR with a seated tracking space that the greyscreen issues occur. I've done a once over on the code and its not immediately obvious what would be have to be done to fix this, hence why I'd rather support you, the active developer, rather than try and hack out a fix from scratch like I'm currently doing with OpenVR Input Emulator.

Let me know if you are interested in the Reverb; you mentioned in another issue that life was keeping you very busy, but hopefully you can find time to work on this since I'm pretty sure soon WMR + knuckles will be the most common use case for this application. 

Thanks!

**Comment by Raivr-dev** [comment]
Hi @JstuffJr ,

I'm not Pushrax, but I'm the developer of another app that aims to improve the VR experience for WMR owners that use Index (Knuckles) controllers. My app (Stop Sign VR) provides a custom boundary system that works for both the WMR headset and the Index controllers. It shows a custom grid when those devices are close to the boundary or ceiling. This feature is currently in beta and will (hopefully) be released in a few days.

I agree that the knuckles + G2 can be a popular combo, and as those users could use my app, I'm interested in understanding any problems associated with that combo.

Can you give a few examples of games that use a seated tracking space and present the grey screen problem with WMR?

**Comment by 4dos** [comment]
Thank you for information and the "sitting game" is "phantom covert ops".
I have a friend who is in a wheelchair. it would be very interesting for you.

**Comment by Raivr-dev** [comment]
Is that a native SteamVR game?

**Comment by 4dos** [comment]
No, it is here:
https://www.oculus.com/experiences/quest/2302118823192509/?locale=de_DE

**Comment by Raivr-dev** [comment]
Thanks. Does the grey screen problem also occur with native SteamVR games?

**Comment by Alde1980** [comment]
> Thanks. Does the grey screen problem also occur with native SteamVR games?

Hi, I still happens with DCS, even DCS steam client.
Some months ago I managed to make work open space calibrator with DCS with WMR reverb and Vive tracker/controllers a couple of times, after that it stopped working and gray screen every time I try to use both systems combined

**Comment by Raivr-dev** [comment]
Is the problem reproducible with the **free** version of _DCS World Steam Edition_?
Any other games?

**Comment by Alde1980** [comment]
To first question yes, reproducible with standalone free client and steam free client.
To second question: It happened with another game but I cannot recall witch one right now. With xplane for example it works as intended and you can sync both playspaces and track the headset with a vive tracker or controller

**Comment by JstuffJr** [comment]
Took a little break there, glad to see all this activity!

@Raivr-dev I am a huge fan of your app, great to hear you are another party working with this. So, my understanding is that SteamVR has two separate tracking universes, separately maintained, for seated and roomscale. The simple way to test which universe a game is using is if the recenter playspace button in the SteamVR menu actually does anything, If it actually recenters, the app should be using the seated tracking universe (and so won't work with OpenVR Space Calibrator + WMR, etc). Otherwise if it doesn't do anything it is in roomscale, as most games are (why people complain all the time about the recenter button not working).

Games I have personally experienced this with are DCS World and Moss... and.... erherm Illusion titles. It is also true Revive appears to use reference off of the seated tracking universe so all those games don't work as well. I'm not an intimate developer with the OpenVR API, so I apologize if any of this is a simplification or wrong, but at least the recenter button trick seems to be pretty reliable. 



@pushrax Absolutely bloody incredible to hear you are interested/working on supporting vive tracker override/reference in your application. It really is the ultimate solution if people don't mind mounting a tracker + the $100 cost, which really shouldn't be an issue I would think with those already spending the money to purchase knuckles with G2 etc. It's very easy to attach a tracker with a simple phone clamp!

![image](https://user-images.githubusercontent.com/5015426/98455335-5c8db300-212c-11eb-9a04-6436040e7367.png)

I have a very tuned room for WMR tracking and still get infuriated with how many minor headtracking glitches it has vs the index/lighthouse, most people just don't notice I guess because they are very minor. Full lighthouse really is the dream. 

Anyways, regarding the Reverb + gray screen, I'll email you once I resolve RMAing my current backup cable. The good 'ol G1 really is an RMA nightmare and you pretty much have to replace the cable every year, with daily use haha.

**Comment by byancey** [comment]
I think I can add Virtual Desktop to this list...or at least the problem I'm seeing with Virtual Desktop using sounds very much like this issue.  The entire screen is grey, buttons, don't responds to bring up steam of Virtual Desktop menus...the WMR boundary does visibly pop-up if I get too close to it.

Interestingly, I have played a couple of revive games with G2 + Knuckles...Drop Dead plays great with this combo via revive, no grey screen issues there.

**Comment by hestenet** [comment]
@pushrax I also have a spare Reverb G1 I'd be willing to ship you, so if it hasn't worked out with the other user feel free to let me know.

**Comment by byancey** [comment]
Just FYI.  I found a solution to this problem that worked for me and is a bit less involved that using a spare Vive tracker.  Details on this post over on reddit:  https://www.reddit.com/r/MixedVR/comments/jzk65t/wmr_hmd_steam_vr_controllers_causes_grey_screen/

**Comment by pushrax** [comment]
To summarize, the fix appears to be to use OpenVR Advanced Settings ability to spoof the SteamVR chaperone. The "Force Use SteamVR Chaperone" setting needs to be enabled, you can find it in this menu https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#--settings-page.

**Comment by AlehandoroVR** [comment]
> To summarize, the fix appears to be to use OpenVR Advanced Settings to turn off a built in SteamVR feature called "enableDriverBoundsImport". Usually SteamVR gets the chaperone bounds from the driver (e.g. WMR driver), but this disables that, so you end up always using the bounds calculated by SteamVR.
> 
> The "Force Use SteamVR Chaperone" setting needs to be enabled, you can find it in this menu https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#--settings-page.
> 
> You also need to disable the WMR chaperone. Before you do this, you'll want to save the working chaperone in OpenVR Advanced Settings https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#--chaperone-page
> 
> I could possibly integrate something that fixes everything at once with Space Calibrator, since it has the ability to store and restore the chaperone too. The enableDriverBoundsImport flag will require one line of code to toggle. Blocking the WMR chaperone could be done with the Space Calibrator driver.

Hello! 

Congratulations for the great work you have done with this application, it is very useful for me.

I would like to know if you finally managed to add an option to make this work without resorting to the other software.

I've been testing some applications and for example Big Screen is another one affected with the grey screen.

I'm currently working with an HP Reverb G2 and the Knuckles.

Thanks a lot!

Translated with www.DeepL.com/Translator (free version)

**Comment by Alde1980** [comment]
Hi AlehandoroVR, I have tested the above mentioned method for a motion compensation rig. I'm from spain too and if tou eant Icanreach you and share my findings speaking in spanish. Just tell me if tou are interested.
Best Regards!

**Comment by AlehandoroVR** [comment]
> Hola AlehandoroVR, he probado el método mencionado anteriormente para una plataforma de compensación de movimiento. Yo también soy de españa y si me quieres puedo contactar contigo y compartir mis hallazgos hablando en español. Dime si te interesa.
> ¡Atentamente!

Sure! why not!

On my Youtube channel you can find my email address or if you have a telegram you can reach me as @alehandorovr

**Comment by Raivr-dev** [comment]
Hi @JstuffJr and others,

I was finally able to investigate the "_grey screen with seated games_" problem. It turns out that the problem is not caused by _SpaceCalibrator_, but it's within SteamVR or WMR-for-SteamVR. I found a solution though, which I implemented in my app, _Stop Sign VR_. I tested with a WMR headset, Index controllers and these seated games:

- DCS World Steam Edition
- Moss
- Apollo 11 VR
- Hellblade: Senua's Sacrifice

They all work flawlessly now.

Steps to take:
1. Use _SpaceCalibrator_ as usual, but don't let it store and restore the chaperone.
2. Turn off _OpenVR Advanced Settings_, or make sure all its options are off where it does something to the chaperone.
3. Install _Stop Sign VR_. Switch to the beta branch. (It will be published to the public branch next week.)
4. Start _StopSignVR_ once manually so it will auto-launch with SteamVR next time. You can turn all features off; it applies the grey screen fix behind the scenes.
5. Exit _SteamVR_.

I start SteamVR from within the WMR Portal by switching on an Index controller. The four mentioned seated games then run fine with WMR + Index controllers. You don't have to do anything special.

Mentioning others who expressed interest in a fix: @hestenet @Alde1980 @4dos @ithirzty @AlehandoroVR .
If you have tested my solution, I'd appreciate to know if all works well.

**Comment by srt23** [comment]
@Raivr-dev Your fix actually works. Kind off. I get an issue where I am transported up in the air as soon as Stop Sign VR starts, and this persist into the games making them unplayable. But games that previously had grey screens now works, like Apollo 11, Transference and Job Simulator. This does not happen if I start Steam VR for wmr from steam, and not by turning on the controllers inside wmr like you suggest. But then I get the grey screen issue again. I should add that I use Vive wands, not index controllers.

**Comment by Raivr-dev** [comment]
Thanks for the report, @srt23.
Does the floating in the air happen every time you start by switching on the Index controller in WMR Portal? Even after a reboot? Could you try turning off the WMR boundaries before starting SteamVR, and then try to turn them on again before starting SteamVR?
If I can reproduce the floating behavior, I will look into it.

**Comment by srt23** [comment]
@Raivr-dev Thank you for the reply. Yes, it happens every time. Makes me float right beneath the roof in the steam vr home defeault environment, with your chaperone beneath me, but with the warning boxes at the right place relative to me. I have tried rebooting and have redone the space calibration a few times. Turning the wmr bounderies on and off doesn't help either. I start at ground level but gets teleported up as soon as Stop Sign VR boots.

**Comment by Raivr-dev** [comment]
@srt23 I uploaded a new version to Steam (on the beta branch). It fixes to floating problem. There's a switch to turn the fix on or off (off needs a restart). The toggle is in _Stop Sign VR_'s dashboard page, on tab _Boundary++_.
Looking forward to your, and others', experience with this beta version (v. 0.10.1.72).

**Comment by srt23** [comment]
@Raivr-dev Thank you. Can confirm it now seems to work. Great job!
